crates/driver/src/infra/cli.rs

```diff
@@ -25,5 +25,10 @@ pub struct Args {
     /// format. For an example see
     /// https://github.com/cowprotocol/services/blob/main/crates/driver/example.toml.
     #[clap(long, env)]
-    pub config: PathBuf,
+    pub config_path: Option<PathBuf>,
+
+    /// Like `config` but reads the configuration directly from the argument
+    /// instead of a file.
+    #[clap(long, env)]
+    pub config_string: Option<String>,
```

Comment:
It's bit unfortunate that this would introduce another state that needs to be handled.
On the one hand I think creating a file before running a test doesn't seem super annoying on the other hand I'm a bit concerned about temporary files not getting cleaned up when a test panics.
Do you think it would make sense to use `#[cfg(test)]` here? Then we could always require the config string to be passed in directly for tests and still expect the path to be passed in for running the binary in prod.
Not sure about that, though.

Reply:
cfg-test doesn't work, because when we depend on the driver crate from the e2e crate, the driver crate is not compiled with cfg-test. It is a normal dependency. Crates are only compiled with cfg-test when their own tests run. 

The extra state is minor imo. It's two more lines in main and this could be useful for the kubernetes setup too. Creating temporary config files on disk is no big deal either, but in general I don't like it when you are forced to create files. I prefer it if e2e tests can be run in memory as much as possible.

Reply:
Personally, I'm not a huge fan of the added flag (my rationale is that we have two flags that do the same thing just for testing). I would have preferred using `tempfile` crate for example for tests (as then the E2E is more "complete" in that it checks configuration deserialization from files that get cleaned up by your OS anyway).

That being said, I don't want to bikeshed too much on this issue.

Reply:
Another way of accomplishing this would be to pass a file descriptor instead of a file string. `tempfile` does give us a descriptor and the descriptor is available to the component we create because it is the same process.

Reply:
> you can run into problems with not picking a unique name, another process deleting the file, the file not getting cleaned up.

I feel like, while `NamedTempFile` is not perfect, I don't expect us to run into these issues practically (as the temp files are very short lived). That being said, running into one of these in CI would be very difficult to debug...

Reply:
We can also use the `NamedTempFile::new_in` in the Cargo `target` dir. There is no reasonable reason why another process would mess with that, and at worst you bloat your `target` dir.